gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala

```diff
@@ -122,40 +123,13 @@ class HiveTableScanExecTransformer(
             ReadFileFormat.JsonReadFormat
           case _ => ReadFileFormat.TextReadFormat
         }
+      case Some(inputFormat)
+          if ORC_INPUT_FORMAT_CLASS.isAssignableFrom(Utils.classForName(inputFormat)) =>
+        ReadFileFormat.OrcReadFormat
       case _ => ReadFileFormat.UnknownFormat
     }
   }
 
-  override protected def doValidateInternal(): ValidationResult = {
-    val validationResult = super.doValidateInternal()
-    if (!validationResult.isValid) {
-      return validationResult
-    }
-
-    val tableMeta = relation.tableMeta
-    val planOutput = output.asInstanceOf[Seq[AttributeReference]]
-    var hasComplexType = false
-    planOutput.foreach(
-      x => {
-        hasComplexType = if (!hasComplexType) {
-          x.dataType.isInstanceOf[StructType] ||
-          x.dataType.isInstanceOf[MapType] ||
-          x.dataType.isInstanceOf[ArrayType]
-        } else hasComplexType
-      })
-
-    fileFormat match {
-      case ReadFileFormat.JsonReadFormat => ValidationResult.ok
-      case ReadFileFormat.TextReadFormat =>
-        if (!hasComplexType) {
-          ValidationResult.ok
-        } else {
-          ValidationResult.notOk("does not support complex type")
-        }
-      case _ => ValidationResult.notOk("Unknown file format")
-    }
```

Comment:
above check is specific for ClickHouse backend so moved to corresponding backend API instead.

Reply:
CC: @zzcclp this piece of code moved to CK backend

Reply:
LGTM